File: app/src/main/java/com/slapshotapps/dragonshockey/models/PlayerStats.java
package com.slapshotapps.dragonshockey.models;

import android.os.Parcel;
import android.os.Parcelable;
import androidx.annotation.Keep;
import androidx.annotation.NonNull;

@Keep
public class PlayerStats implements Comparable<PlayerStats>, Parcelable {

    public final int playerID;
    public final String firstName;
    public final String lastName;
    public final PlayerPosition position;
    public int goals;
    public int assists;
    public int gamesPlayed;
    public int points;
    public int penaltyMinutes;
    public int goalsAgainst;
    public int shutouts;


    public PlayerStats(int playerID, String firstName, String lastName, PlayerPosition position) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.playerID = playerID;
        this.position = position;
    }

    @Override
    public int compareTo(@NonNull PlayerStats playerStats) {

        if (points < playerStats.points) {
            return 1;
        } else if (points > playerStats.points) {
            return -1;
        }
        return lastName.compareTo(playerStats.lastName);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(this.playerID);
        dest.writeString(this.firstName);
        dest.writeString(this.lastName);
        dest.writeInt(this.position == null ? -1 : this.position.ordinal());
        dest.writeInt(this.goals);
        dest.writeInt(this.assists);
        dest.writeInt(this.gamesPlayed);
        dest.writeInt(this.points);
        dest.writeInt(this.penaltyMinutes);
        dest.writeInt(this.goalsAgainst);
        dest.writeInt(this.shutouts);
    }

    protected PlayerStats(Parcel in) {
        this.playerID = in.readInt();
        this.firstName = in.readString();
        this.lastName = in.readString();
        int tmpPosition = in.readInt();
        this.position = tmpPosition == -1 ? null : PlayerPosition.values()[tmpPosition];
        this.goals = in.readInt();
        this.assists = in.readInt();
        this.gamesPlayed = in.readInt();
        this.points = in.readInt();
        this.penaltyMinutes = in.readInt();
        this.goalsAgainst = in.readInt();
        this.shutouts = in.readInt();
    }

    public static final Creator<PlayerStats> CREATOR = new Creator<PlayerStats>() {
        @Override
        public PlayerStats createFromParcel(Parcel source) {
            return new PlayerStats(source);
        }

        @Override
        public PlayerStats[] newArray(int size) {
            return new PlayerStats[size];
        }
    };
}

File: app/src/main/java/com/slapshotapps/dragonshockey/models/PlayerStats.kt
package com.slapshotapps.dragonshockey.models

import android.os.Parcel
import android.os.Parcelable
import androidx.annotation.Keep

@Keep
class PlayerStats(val playerID: Int, val firstName: String, val lastName: String, val position: PlayerPosition) :
        Comparable<PlayerStats>, Parcelable {
    var goals: Int = 0

    var assists: Int = 0

    var gamesPlayed: Int = 0

    var points: Int = 0

    var penaltyMinutes: Int = 0

    var goalsAgainst: Int = 0

    var shutouts: Int = 0

    override fun compareTo(other: PlayerStats): Int {

        if (points < other.points) {
            return 1
        } else if (points > other.points) {
            return -1
        }
        return lastName.compareTo(other.lastName)
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<PlayerStats> = object : Parcelable.Creator<PlayerStats> {
            override fun createFromParcel(source: Parcel): PlayerStats = PlayerStats(source)
            override fun newArray(size: Int): Array<PlayerStats?> = arrayOfNulls(size)
        }
    }

    constructor(source: Parcel) : this(
            source.readInt(),
            source.readString(),
            source.readString(),
            PlayerPosition.values()[source.readInt()]
    ) {
        this.goals = source.readInt();
        this.assists = source.readInt();
        this.gamesPlayed = source.readInt();
        this.points = source.readInt();
        this.penaltyMinutes = source.readInt();
        this.goalsAgainst = source.readInt();
        this.shutouts = source.readInt();
    }


    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) = with(dest) {
        writeInt(playerID)
        writeString(firstName)
        writeString(lastName)
        writeInt(position.ordinal)
        writeInt(goals)
        writeInt(assists)
        writeInt(gamesPlayed)
        writeInt(points)
        writeInt(penaltyMinutes)
        writeInt(goalsAgainst)
        writeInt(shutouts)
    }
}
